BaragonUI/app/components/groupDetail/GroupDetail.jsx

```diff
@@ -2,35 +2,318 @@ import React, { Component, PropTypes } from 'react';
 import { connect } from 'react-redux';
 import classNames from 'classnames';
 import { Link } from 'react-router';
+
+import UITable from '../common/table/UITable';
+import Column from '../common/table/Column';
+import AddTrafficSourceButton from '../common/modalButtons/AddTrafficSourceButton';
+import RemoveTrafficSourceButton from '../common/modalButtons/RemoveTrafficSourceButton';
+import RemoveKnownAgentButton from '../common/modalButtons/RemoveKnownAgentButton';
+import ModifyCountButton from '../common/modalButtons/ModifyCountButton';
+
+import { refresh } from '../../actions/ui/groupDetail'
+
 import rootComponent from '../../rootComponent';
-import { refresh } from '../../actions/ui/groupDetail';
 import Utils from '../../utils';
 
 class GroupDetail extends Component {
-  render() {
-    // TODO - nicely render all of the group details + wire up actions
-    // - add/remove traffic source (requires new modal)
-    // - json button, use existing component
-    // - remove known agent (requires new modal)
-    // - remove base path (requires new modal)
-    return <h1>Group Detail</h1>
-  }
-
   static propTypes = {
+    domain: React.PropTypes.string,
+    domainsServed: React.PropTypes.array,
     params: PropTypes.object.isRequired,
-    group: React.PropTypes.object,
+    group: React.PropTypes.string,
     basePaths: React.PropTypes.array,
     targetCount: React.PropTypes.number,
     agents: React.PropTypes.array,
-    knownAgents: React.PropTypes.array
+    knownAgents: React.PropTypes.array,
+    editable: React.PropTypes.bool
   };
+
+  render() {
+    return (
```

Comment:
the general html generated by this is fine, but a comment on the organization here. Right now this is a lot of code in one component. I would suggest breaking things up into smaller components (that can also live in this same folder). For example, you could have a `<TrafficSources>` component that takes a prop of the traffic source data. You could do similar for an Agents or a KnownAgents component or a BasePaths component.

Not only does this help us to read and organize things a bit better, it also means we can reuse things like that elsewhere in the ui if we need to.

for example a snippet of the code would read more like:
```
<div className="row">
  <TrafficSources 
      trafficSources={this.props.trafficSources}
      group={this.props.group}
  />
</div>
```